Tests/MapboxNavigationTests/CarPlayManagerSpec.swift

```diff
@@ -0,0 +1,206 @@
+                    let mapTemplateSpy = carPlayManager.interfaceController?.topTemplate as? MapTemplateSpy
+                    
+                    expect(mapTemplateSpy?.currentTripPreviews).toNot(beEmpty())
+                    expect(mapTemplateSpy?.currentPreviewTextConfiguration?.startButtonTitle).to(equal(customStartButtonTitleText))
+                }
+            }
+        }
+        
+        describe("Starting a valid trip.") {
+            let action = {
+                let routeChoice = createValidRouteChoice()
+                let trip = createTrip(routeChoice)
+                let mapTemplate = CPMapTemplate()
+                
+                carPlayManager.mapTemplate(mapTemplate, startedTrip: trip, using: routeChoice)
+                carPlayManager.carPlayNavigationViewController?.loadViewIfNeeded()
+                
+                let navigationService = delegate.currentService as? MapboxNavigationService
+                navigationService?.start()
+            }
+            
+            context("When configured to simulate.") {
+                beforeEach {
+                    carPlayManager.simulatesLocations = true
+                    carPlayManager.simulatedSpeedMultiplier = 5.0
+                }
+                
+                it("Starts navigation with a navigation service with simulation enabled.") {
+                    action()
+                    
+                    expect(delegate.navigationInitiated).to(beTrue())
+                    let navigationService = delegate.currentService as? MapboxNavigationService
+                    
+                    expect(navigationService?.simulationMode).to(equal(.always))
+                    expect(navigationService?.simulationSpeedMultiplier).to(equal(5.0))
+                }
+            }
+            
+            context("When configured not to simulate.") {
+                beforeEach {
+                    carPlayManager.simulatesLocations = false
+                }
+                
+                it("Starts navigation with a navigation service with simulation set to inTunnels by default.") {
+                    action()
+                    
+                    expect(delegate.navigationInitiated).to(beTrue())
+                    let navigationService = delegate.currentService as? MapboxNavigationService
+                    
+                    expect(navigationService?.simulationMode).to(equal(.inTunnels))
+                }
+            }
+        }
+        
+        describe("Starting an invalid trip.") {
+            context("Precondition should be triggered if CPRouteChoice is invalid.") {
+                let routeChoice = createInvalidRouteChoice()
+                let trip = createTrip(routeChoice)
+                let mapTemplate = CPMapTemplate()
+                
+                expect {
+                    carPlayManager.mapTemplate(mapTemplate, startedTrip: trip, using: routeChoice)
```

Comment:
In case if uses passes `CPRouteChoice`, which has `userInfo` set to type different than `CarPlayUserInfo` - assert.